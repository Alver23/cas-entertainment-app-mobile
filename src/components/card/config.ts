export enum STATUS_TYPES {
  GREEN = 'green',
  BLUE = 'blue',
}
