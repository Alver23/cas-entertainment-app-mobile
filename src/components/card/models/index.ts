// Dependencies
import { ViewProps } from 'react-native';

export interface ICard {
  backgroundImage: ViewProps;
  onPress: () => void;
  text: string;
}
