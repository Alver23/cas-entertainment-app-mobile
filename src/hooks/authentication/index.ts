// Dependencies
import React from 'react';

// Contexts
import { AuthContext } from '@contexts/auth/auth-context';
import { IAuthContext } from '@contexts/auth/auth-context-interface';

export const useAuthentication = () => {
  return React.useContext<IAuthContext>(AuthContext);
};
