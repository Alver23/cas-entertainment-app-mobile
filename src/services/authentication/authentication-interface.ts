// Models
import { IUserAuth } from '@models/user-interface';

export interface ILoginRequest {
  email: string;
  password: string;
}

export type ILoginResponse = IUserAuth;
