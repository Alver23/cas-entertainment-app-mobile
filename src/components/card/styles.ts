// Hooks
import { makeStyles } from '@hooks/theme';

// Models
import { ITheme } from '@core/theme';

import { STATUS_TYPES } from './config';

const styles = makeStyles(({ colors }: ITheme) => ({
  container: {
    flex: 1,
    borderRadius: 8,
    borderWidth: 1,
    margin: 4,
    padding: 8,
  },
  [STATUS_TYPES.GREEN]: {
    borderColor: colors.primary,
  },
}));

export default styles;
