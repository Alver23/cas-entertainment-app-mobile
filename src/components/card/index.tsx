// Dependencies
import React, { ReactElement } from 'react';
import { TouchableOpacity } from 'react-native';

// Models
import { ICard } from './models';

// Styles
import useStyles from './styles';

const Card = ({ status, children, onPress }: ICard): ReactElement => {
  const styles = useStyles();
  return (
    <TouchableOpacity onPress={onPress} style={[styles.container, styles[status]]}>
      {children}
    </TouchableOpacity>
  );
};

export default Card;
