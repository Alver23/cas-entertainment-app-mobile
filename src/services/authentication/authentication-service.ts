// Core
import { externalAxiosInstance } from '@core/axios';

// config
import endpoints from '@config/endpoints';

// Connector
import { Connector } from '@core/connector';

// Models
import { ILoginRequest, ILoginResponse } from './authentication-interface';

export class AuthenticationService {
  constructor(private readonly http: Connector) {}

  public login(payload: ILoginRequest): Promise<ILoginResponse> {
    const url = endpoints.login;
    const { email, password } = payload;
    const auth = {
      password,
      username: email,
    };
    return this.http.post(url, {}, { auth });
  }
}

export const authenticationService = new AuthenticationService(new Connector(externalAxiosInstance));
