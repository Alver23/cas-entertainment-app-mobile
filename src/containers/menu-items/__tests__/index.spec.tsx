// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import * as ReactNavigationHooks from '@react-navigation/native';

// Hooks
import * as AuthenticationHooks from '@hooks/authentication';

// Components
import { MemoizedCard } from '@components/card';

import { MenuItems } from '../index';

// Mocks
import mocks from './mocks.json';

jest.mock('@react-navigation/native');

describe('Menu Items container', () => {
  let component: ShallowWrapper;

  const navigation = {
    navigate: jest.fn(),
  };

  beforeEach(() => {
    jest.spyOn(ReactNavigationHooks, 'useNavigation').mockReturnValue(navigation);
    jest.spyOn(AuthenticationHooks, 'useAuthentication').mockReturnValue(mocks);
    component = shallow(<MenuItems />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should navigate to screen when on press menu item', () => {
    const onPress = component.find(MemoizedCard).first().prop('onPress');
    onPress();
    expect(navigation.navigate).toHaveBeenCalled();
  });
});
