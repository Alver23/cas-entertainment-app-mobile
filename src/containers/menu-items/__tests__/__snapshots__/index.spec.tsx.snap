// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Items container should save a snapshot of the component 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <Memo(Card)
    key="1"
    onPress={[Function]}
  />
</View>
`;
